ndd_slurm.py

```diff
@@ -196,15 +228,31 @@ def get_ssh_cmds(args):
 
 
 def run_ssh_master(args):
-    cmds = [get_master_ndd_cmd(args)] + get_ssh_cmds(args)
+    if args.r:
+        read_fd, write_fd = os.pipe()
+        tar = init_process(['tar', '-c', args.i], write_fd=write_fd)
+    else:
+        read_fd = None
+    cmds = [get_master_ndd_cmd(args, read_fd)] + get_ssh_cmds(args)
     procs = {proc.pid: proc for proc in map(init_process, cmds)}
+    if args.r:
```

Comment:
Let's move this to actual creation of tar process. Also it seems to be simpler if we change `wait` to use a `list` rather than `dict`:
```
procs = []
if args.r:
  ...
  procs.append(init_process(...))
else:
  ...
procs.extend(map(init_process(cmds))
return wait(procs)

def wait(procs):
  proc_map = {proc.pid: proc for proc in procs}
  ...
```